### PR TITLE
[ray 2.0] adjust the release test of ray lightning master

### PR DESCRIPTION
**Pull request by JiahaoYao**
## Why are these changes needed?

- [x] adding the test_master file because master and latest branch diverge. 
- [x] adding the test in the release test 

## Related issue number

previously, the ray lightning update breaks the master of ray. The reason is some of the test in the ray library about the ray lightning is old. Here are update the version of the ray lightning here. 

previous issue: 
- https://github.com/ray-project/ray/issues/27274

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ml_user_tests/ray-lightning/ray_lightning_user_test.py [hunk 1]
```diff
@@ -3,7 +3,14 @@
...
 import ray
-from simple_example import main
+import sys
+
+# add a switch here because master and latest branch
+# diverge; this will be called in `release/release_tests.yaml`
+if len(sys.argv) > 1 and sys.argv[1] == "master":
+    from simple_example_master import main
+else:
+    from simple_example import main
```
Comment (matthewdeng) [inline release/ml_user_tests/ray-lightning/ray_lightning_user_test.py:13]: Can we move this check to `simple_example.main` and delay imports?

Something like:

```python
if use_master:
    # ray_lightning >= 0.3.0
    from ray_lightning import RayStrategy

    trainer = pl.Trainer(
        strategy=RayStrategy(num_workers=num_workers, use_gpu=use_gpu),
        max_steps=max_steps,
    )
else:
    # ray_lightning < 0.3.0
    from ray_lightning import RayPlugin

    trainer = pl.Trainer(
        plugins=[RayPlugin(num_workers=num_workers, use_gpu=use_gpu)],
        max_steps=max_steps,
    )
```

Comment (matthewdeng) [reply release/ml_user_tests/ray-lightning/ray_lightning_user_test.py:13]: If you want to make this a bit cleaner and support the case when we release `0.3.0`, you can use `packaging.version` to compare versions, though I'm not sure what the version number is for master.

**Comment by JiahaoYao** [comment]
smart!

File: release/ml_user_tests/ray-lightning/ray_lightning_user_test.py (hunk 1 above)
Comment (JiahaoYao) [reply release/ml_user_tests/ray-lightning/ray_lightning_user_test.py]: thanks @matthewdeng , changed! this is smart!

**Review by JiahaoYao — COMMENTED** [review]
ready for merge!

**Comment by JiahaoYao** [comment]
sorry all! let me fix this

**Pull request closed without merging**